spec/support/page_helpers.rb

```diff
@@ -2,22 +2,26 @@
 
 # Used for testing background jobs in integration tests where we need to reload the page until the job is done.
 module PageHelpers
-  def reload_page_until_timeout(timeout: 10)
+  TRIES = 10
+
+  def reload_page_until_timeout
     browser = page.driver.browser
     content = nil
-    Timeout.timeout(timeout) do
```

Comment:
The `Timeout` library caused non-deterministic test failures, and sometimes got stuck in endless loops. `Timeout` is bad because it doesn't play nicely in threaded environments (see, e.g., https://adamhooper.medium.com/in-ruby-dont-use-timeout-77d9d4e5a001). So I ditched the old approach and came up with a simpler one that plays nice with threads.